test/test_functions

```diff
@@ -1181,3 +1193,195 @@ cp_usrbin() {
     sudo chmod 0444 $f || return 3
   done
 }
+
+
+# prints a benchmark log message
+benchmark_log() {
+  local log_message=$1
+  echo "===========    ${log_message}"
+}
+
+
+# waits until cvmfs is mounted
+wait_fqrn_mount() {
+  local waits=0
+  while ! mount | grep -q "/cvmfs/${FQRN}"; do
+    waits=$(( waits + 1 ))
+    if [ "$waits" -ge 15 ]; then
+      die "${FQRN} could not be mounted"
+    fi
+    sleep 1
+  done
+}
+
+
+# runs the command once in order to get a warm cache
+create_warm_cache() {
+  local workspace=$1
+  cvmfs2 -f -o config=cvmfs.conf,disable_watchdog,simple_options_parsing $FQRN /cvmfs/$FQRN 2>&1 & #> /dev/null 2>&1 &
+  wait_fqrn_mount
+
+  cvmfs_run_benchmark 2>&1
+  cvmfs_umount $FQRN
+  return 0
+}
+
+
+# creates the default cvmfs.conf file for normal testing
+#
+# @param destination_file  absolute path of the directory
+create_default_cvmfs_config() {
+  local destination_directory="$1"
+  local config_file=$destination_directory/cvmfs.conf
+  local test_folder=$( cd $(dirname $0) ; pwd -P )
+
+  cat > $config_file << EOF
+CVMFS_CACHE_BASE=${CVMFS_OPT_RAMCACHE}
+CVMFS_RELOAD_SOCKETS=${CVMFS_OPT_RAMCACHE}
+CVMFS_SERVER_URL=http://cvmfs-stratum-one.cern.ch/cvmfs/@fqrn@
+CVMFS_SHARED_CACHE=no
+CVMFS_HTTP_PROXY=${CVMFS_TEST_PROXY}
+CVMFS_AUTO_UPDATE=false
+CMS_LOCAL_SITE=${test_folder}/benchmark_tests/004-cms
+CVMFS_QUOTA_LIMIT=-1
+
+EOF
+
+}
+
+
+# sets the valgrind options depending on the test type
+#
+# @param test_type  concrete test to execute
+set_valgrind_options() {
+  local CVMFS_OPT_CALLGRIND="--tool=callgrind --dump-before=fuse_session_loop_mt --dump-after=fuse_session_loop_mt --dsymutil=yes --log-file=${FQRN}.callgrind.log -v"
+  local CVMFS_OPT_MEMCHECK="--tool=memcheck --xml=yes --xml-file=${FQRN}.memcheck.xml --trace-children=yes --child-silent-after-fork=yes --leak-check=yes --log-file=${FQRN}.memcheck.log -v"
+  OPTIONS=""
+  case $CVMFS_OPT_TEST_TYPE in
+    callgrind)
+      OPTIONS=$CVMFS_OPT_CALLGRIND
+    ;;
+    memcheck)
+      OPTIONS=$CVMFS_OPT_MEMCHECK
+    ;;
+  esac
+}
+
+
+# creates the environment to run the benchmark
+#
+# @param workspace  absolute path to the directory where the benchmark will put the files in
+setup_benchmark_environment() {
+  local workspace="$1"
+  mkdir -p /cvmfs/$FQRN
+  mkdir -p $CVMFS_OPT_OUTPUT_DIR/$FQRN
+  mkdir -p $workspace/cache
+
+  # cleaning the cache
+  rm -rf ${CVMFS_OPT_RAMCACHE}/${FQRN}/*
+
+  # mtab trick
+  if [ ! -h /etc/mtab ]; then
+    mv /etc/mtab /etc/mtab.save
+    ln -s /proc/mounts /etc/mtab
+  fi
+
+  # check that the cache directory exists, is empty and is correctly labeled for SElinux
+  if [ ! -d $CVMFS_OPT_RAMCACHE ]; then
+    mkdir -p $CVMFS_OPT_RAMCACHE
+    has_selinux
+    selinux=$?
+    if [ "$selinux" -eq 0 ]; then
+      chcon -Rv --type=cvmfs_cache_t $CVMFS_OPT_RAMCACHE
+    fi
+  else
+    rm -rf $CVMFS_OPT_RAMCACHE/*
```

Comment:
The point of checking here is to prevent a disaster even when the code gets restructured later.  This checking should be done right before every `rm -rf ${VARIABLE}/*`.  The problem of the undefined $VARIABLE is so common, it probably happened to everybody at least once.